src/vs/editor/contrib/smartSelect/common/smartSelect.ts

```diff
@@ -196,3 +198,84 @@ class ShrinkSelectionAction extends AbstractSmartSelect {
 		});
 	}
 }
+
+@editorAction
+class SelectBracketAction extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.expandSelectionToBrackets',
+			label: nls.localize('smartSelect.expandSelectionToBrackets', "Expand Selection To Brackets"),
+			alias: 'Expand Selection To Brackets',
+			precondition: null,
+			kbOpts: {
+				kbExpr: EditorContextKeys.textFocus,
+				primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.US_SLASH
+			}
+		});
+	}
+
+	public run(accessor: ServicesAccessor, editor: ICommonCodeEditor): void {
+		const model = editor.getModel();
+		if (!model) {
+			return;
+		}
+
+		const selection = editor.getSelection();
+		if (!selection.isEmpty()) {
+			return;
+		}
+
+		const originalPosition = selection.getStartPosition();
+
+		// find previous open bracket with no match
+		let prevBracket = null;
+		let currentPosition = new Position(originalPosition.lineNumber, originalPosition.column);
+		let setCount = 0;
+		do {
+			prevBracket = model.findPrevBracket(currentPosition);
+			if (prevBracket) {
+				currentPosition = new Position(prevBracket.range.startLineNumber, prevBracket.range.startColumn);
+				!prevBracket.isOpen ? setCount++ : setCount--;
+			}
+		} while (prevBracket && !(prevBracket.isOpen && setCount < 0));
```

Comment:
If we use `findPrevBracket` for bracket searching, then it will catch all possible types of brackets `{ [ (` but a single `setCount` can't handle all types.